reactor-core/src/main/java/reactor/core/publisher/FluxFilter.java

```diff
@@ -206,7 +215,13 @@ public void onNext(T t) {
 				b = predicate.test(t);
 			}
 			catch (Throwable e) {
-				onError(Operators.onOperatorError(s, e, t, actual.currentContext()));
+				Throwable e_ = Operators.onNextFailure(t, e, actual.currentContext(), s);
```

Comment:
onNextFailure should encapsulate the whole thing ( if then cancel+error else request ) being passed a CoreSubscriber.